.github/workflows/unit-test.yml

```diff
@@ -47,6 +47,55 @@ jobs:
     - name: Running the tests
       run: npm run test-unit -- --ci --maxWorkers=2 --cacheDirectory="$HOME/.jest-cache"
 
+  unit-js-date:
+    name: JavaScript / Date functions
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        timezone: ['EST', 'GMT', 'CET']
+        locale: ['en_US', 'ja_JP']
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Cache node modules
+      uses: actions/cache@v2
+      env:
+        cache-name: cache-node-modules
+      with:
+        # npm cache files are stored in `~/.npm` on Linux/macOS
+        path: ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+
+    - name: Use Node.js 14.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 14.x
+
+    - name: Npm install and build
+      # It's not necessary to run the full build, since Jest can interpret
+      # source files with `babel-jest`. Some packages have their own custom
+      # build tasks, however. These must be run.
+      run: |
+        npm ci
+        npx lerna run build
+
+    - name: Running the tests
+      env:
+        TZ: ${{ matrix.timezone }}
+        LANG: ${{ matrix.locale }}
```

Comment:
Is there a way to achieve the same thing without creating a big number of jobs? Maybe by hooking into jsdom or something? We have limited resources on the WP org github running and this is going to hurt :(

Reply:
The other  "easy" alternative is to just do the same thing that is done here but using a loop after the end of the unit-test job (without creating new jobs).

Reply:
One more (easy) alternative that comes to mind: We could limit these tests to be only run if files in `packages/date/` are touched :thinking:

Reply:
We'd miss package upgrades and things like that and if we do include package-lock.json they'd be triggering very often.

Reply:
How about a bash file or npm script which these tests?

```bash
#!/bin/bash

timezones=(EST GMT CET)
locales=(en_US ja_JP)

for timezone in "${timezones[@]}"; do
    for locale in "${locales[@]}"; do
        ( TZ=$timezone LANG=$locale npm run test-unit -- packages/date )
    done
done
```

npm script would become horribly long:
```
"unit-test:date": "TZ=EST LANG=en_US npm run test-unit -- packages/date && TZ=CET LANG=en_US npm run test-unit -- packages/date ......."

Reply:
Draft here: https://github.com/WordPress/gutenberg/pull/27600